src/main/scala/edu/berkeley/ce/sparkrocks/Block.scala

```diff
@@ -532,41 +571,70 @@ case class Block(center: Array[Double], faces: Seq[Face], generation: Int=0) ext
         allVertices.map { v => v(2) }.sum / allVertices.size
       )
     } else {
-      Array(
-        // Factor of 3 comes from: centroid / (2.0 * (volume/6.0))
-        3.0 * centroidX / totalVolume,
-        3.0 * centroidY / totalVolume,
-        3.0 * centroidZ / totalVolume
-      )
+      val increments = faces.flatMap { face =>
+        // First vertex of face used in all iterations
+        val anchorVertex = vertices(face)(0)
+
+        // Tetrahedra are constructed using three vertices at a time
+        vertices(face).takeRight(vertices(face).length - 1).sliding(2).map { faceVertices =>
```

Comment:
Yeah, don't know why I didn't just write that the first time round. I think leaving it as `(0)` is better for consistency since there's a bunch of `(1)` and `(2)` in there too, if that makes sense.